### PR TITLE
Log the launch command for Spark daemons

### PR DESCRIPTION
**Pull request by pwendell**
For debugging and analysis purposes, it's nice to have the exact command
used to launch Spark contained within the logs. This adds the necessary
hooks to make that possible.

### CODE REVIEW

**Comment by rxin** [comment]
Is this only for Spark daemons?

When I filed the ticket, I think I might've been referring to the applications too.

**Comment by pwendell** [comment]
@rxin Good idea. I added this now for the executors as well.

File: core/src/main/scala/spark/deploy/worker/ExecutorRunner.scala [hunk 1]
```diff
@@ -161,9 +164,15 @@ private[spark] class ExecutorRunner(
       env.put("SPARK_LAUNCH_WITH_SCALA", "0")
       process = builder.start()
 
+      val header = "Spark Executor Command: %s\n%s\n\n".format(command.mkString(" "), "=" * 40)
```
Comment (rxin) [inline core/src/main/scala/spark/deploy/worker/ExecutorRunner.scala]: To make the command re-runnable, let's wrap all parts of the command with quotes.

**Comment by pwendell** [comment]
Hey @rxin this prints the arguments with quotes (and escapes any existing quotes in them).

**Comment by pwendell** [comment]
Jenkins, test this please.

**Pull request merged**